src/hooks/useWindowDimensions/index.ts

```diff
@@ -92,4 +101,35 @@ export default function (useCachedViewportHeight = false): WindowDimensions {
         isExtraSmallScreenWidth,
         isSmallScreen,
     };
+
+    if (!lockedWindowDimensionsRef.current && !isFullScreenRef.current) {
+        return windowDimensions;
+    }
+
+    const didScreenChangeOrientation =
+        isMobile &&
+        lockedWindowDimensionsRef.current &&
+        isExtraSmallScreenWidth === lockedWindowDimensionsRef.current.isExtraSmallScreenWidth &&
+        isSmallScreenWidth === lockedWindowDimensionsRef.current.isSmallScreen &&
+        isMediumScreenWidth === lockedWindowDimensionsRef.current.isMediumScreenWidth &&
+        isLargeScreenWidth === lockedWindowDimensionsRef.current.isLargeScreenWidth &&
+        lockedWindowDimensionsRef.current.windowWidth !== windowWidth &&
+        lockedWindowDimensionsRef.current.windowHeight !== windowHeight;
+
+    // if video is in fullscreen mode, lock the window dimensions since they can change and casue whole app to re-render
+    if (!lockedWindowDimensionsRef.current || didScreenChangeOrientation) {
+        lockWindowDimensions(windowDimensions);
+        return windowDimensions;
+    }
+
+    const didScreenReturnToOriginalSize = lockedWindowDimensionsRef.current.windowWidth === windowWidth && lockedWindowDimensionsRef.current.windowHeight === windowHeight;
```

Comment:
Unlocking window dimensions would not happen under edge conditions, which occur when `lockedWindowDimensionsRef` and `current` window dimensions do not match thereby causing #50818